### PR TITLE
Add "sources" option support for library.

### PR DESCRIPTION
**Pull request by NexushasTaken**
Currently, files in libraries must be in the root folder next to the manifest.json.
This is cumbersome when organizing the files in this folder.  

Although, library system still in early alpha, but still, adding this feature would be great.

### CODE REVIEW

File: src/build/libraries.c [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #define MANIFEST_FILE "manifest.json"
 
 const char *manifest_default_keys[][2] = {
+		{"sources", "Paths to library sources for targets."},
```
Comment (lerno) [inline src/build/libraries.c:7]: Maybe add: "such as interface files"

File: src/build/libraries.c [hunk 2]
```diff
@@ -291,7 +297,16 @@ void resolve_libraries(BuildTarget *build_target)
...
 			vec_add(build_target->ccompiling_libraries, target);
 		}
-		file_add_wildcard_files(&build_target->sources, library->dir, false, c3_suffix_list, 3);
+		if (target->source_dirs) {
+			const char **files = target_expand_source_names(library->dir, target->source_dirs, c3_suffix_list, &build_target->object_files, 3, true);
+			FOREACH(const char *, file, files)
+			{
+				vec_add(build_target->sources, file);
+			}
+		} else {
```
Comment (lerno) [inline src/build/libraries.c:306]: Can you update the brace style to match the surrounding code?

**Review by lerno — CHANGES_REQUESTED** [review]
Please have a look at those 2 minor changes.

File: src/build/libraries.c (hunk 1 above)
Comment (NexushasTaken) [reply src/build/libraries.c]: Done.

File: src/build/libraries.c (hunk 2 above)
Comment (NexushasTaken) [reply src/build/libraries.c]: Done.

**Comment by lerno** [comment]
Oh, I forgot – can you add a note in the releasenotes.md as well?

**Comment by NexushasTaken** [comment]
Apologies i made a mistake rebasing, but I’ve fixed it right away.

but, I've added the changes to `releasenotes.md`

**Comment by lerno** [comment]
Thank you!

**Pull request merged**